sda-commons-server-s3-testing/src/main/java/org/sdase/commons/server/s3/testing/S3Mock.java

```diff
@@ -0,0 +1,121 @@
+package org.sdase.commons.server.s3.testing;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.List;
+import org.sdase.commons.server.s3.testing.builder.MockObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3Mock {
```

Comment:
Make abstract? If we also rename it to `AbstractS3Mock` we wouldn't have a name clash with the "internalS3Mock" field.

Reply:
Should name if `Abstract` although it is not abstract? Or did you also mean changing the class being abstract as well?

Reply:
We cannot make the class abstract as it is being instantiated in the JUnit 4 rule class. Leaving it as it is.

Reply:
👍  also because it's not the base class of the rule and the extensions. That's what I initially thought.